### ISSUE TITLE
Optional params in extended DTO marked required

### ISSUE THREAD

**Issue by auerbachb** [nestjs/swagger#1062]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

I have 2 DTOs the parent DTO has only optional properties, the child DTO inherits from the parent, NestJS recognizes the parent DTO property is optional and doesn't throw an error, Swagger shows the properties as required and throws an error (this matters to me because I'm using the OpenAPI generator to make the client code).
(I cannot use `PartialType` in the inheritance because that breaks the global type coercion as documented in [this issue](https://github.com/nestjs/nest/issues/5844))

```javascript
import { IsNumber, IsOptional, IsPositive, Min } from 'class-validator';

export class PaginationQueryDto {
  @IsOptional()
  @IsNumber()
  limit: number;

  @IsOptional()
  @IsNumber()
  offset: number;
}
```

```javascript
import { IsNumber, IsOptional } from 'class-validator';

import { ApiProperty } from '@nestjs/swagger';

import { PaginationQueryDto } from '../../common/dto/pagination-query.dto';

export class OrdersQueryDto extends PaginationQueryDto {
  @IsOptional()
  @IsNumber()
  readonly show_id: number;
}

```

![image](https://user-images.githubusercontent.com/4504560/100780838-18a26c80-33bf-11eb-97a7-922fefadb0d2.png)

actual API call _** no problems at all**_

![image](https://user-images.githubusercontent.com/4504560/100781126-7df65d80-33bf-11eb-81a2-d01cb9980df7.png)

## Environment

<pre><code>
Nest version: X.Y.Z

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by auerbachb** [comment]
I figured this case out. I expected that query params would be treated the same as the body, ie that the generated OpenAPI file / SwaggerUI would pick up the `class-validation` decorators; however, I see this is not the case; thus, I simply added `@ApiQuery` with `required: false` to resolve the issue; perhaps this could be documented better? 

![image](https://user-images.githubusercontent.com/4504560/100788969-ae8fc480-33ca-11eb-9687-ffc0075a6d2f.png)

since the `@Body` parsing includes all the `class-validator` decorators e.g. `@IsOptional()`; and it does pick up the query params from `@Query` but it doesn't support `class-validator` decorators here.
